src/gpp/GppManager.js

```diff
@@ -0,0 +1,151 @@
+import { CmpApi, CmpStatus, SignalStatus } from '@iabgpp/cmpapi';
+import { debug, getCookieDomain } from '../shared/utils';
+import { loadStub } from './stub/GppStub';
+import {
+	NOTICE_FIELDS,
+	OPT_OUT_FIELDS,
+	SENSITIVE_DATA_PROCESSING_LENGTH_FIELD,
+	SENSITIVE_DATA_PROCESSING_LENGTHS,
+	KNOWN_CHILD_SENSIITIVE_DATA_FIELD,
+	KNOWN_CHILD_SENSIITIVE_DATA_LENGTHS,
+	US_MSPA_FIELDS
+} from './GppAPIUtils'
+import Cookies from "js-cookie";
+
+const GPP_STRING_COOKIE_NAME = 'gpp'
+const CONSENT_VALUES = {
+	notApplicable: 0,
+	optOut: 1,  // opt-out or no consent
+	consent: 2,  // no opt-out or consent
+}
+const NOTICE_VALUES = {
+	notApplicable: 0,
+	provided: 1,
+	notProvided: 2,
+}
+const MSPA_FIELDS_VALUES = {
+	notApplicable: 0,
+	yes: 1,
+	no: 2,
+}
+
+class GppManager {
+
+	constructor(options) {
+		this.section = `us${options.region}v1`;
+		this.gppApplies = options.gppApplies;
+		this.isSubjectToGPP = options.isSubjectToGPP;
+		this.cookieAttributes = {
+			domain: getCookieDomain(window.location.hostname),
+			expires: 365 // 1 year
+		};
+
+		// Install temporary stub until full CMP will be ready
+		if (typeof window.__gpp === 'undefined' && this.gppApplies) {
+			GppManager.installStub();
+		}
+	}
+
+	static installStub() {
+		loadStub();
+	}
+
+	setup() {
+		if (!this.gppApplies) {
+			debug('GPP', 'Geo does not require API');
+		} else {
+			debug('GPP', 'Geo requires API');
+			this.cmpApi = new CmpApi(1, 3);
```

Comment:
when I look at this, I see one major problem - we will load GPP's CmpApi all the time, even if GPP is not required... it will be problematic in GDPR, I would love to dynamically import CmpApi 
```suggestion
            
            const { CmpApi, CmpStatus } = await import(/* webpackChunkName: "iabgpp-cmpapi" */ '@iabgpp/cmpapi');
			this.cmpApi = new CmpApi(1, 3);
```